src/video/windows/SDL_windowskeyboard.c

```diff
@@ -130,20 +128,44 @@ void WIN_UpdateKeymap(SDL_bool send_event)
         }
 
         /* If this key is one of the non-mappable keys, ignore it */
-        /* Uncomment the second part re-enable the behavior of not mapping the "`"(grave) key to the users actual keyboard layout */
-        if ((keymap[scancode] & SDLK_SCANCODE_MASK) /*|| scancode == SDL_SCANCODE_GRAVE*/) {
+        /* Uncomment the third part to re-enable the behavior of not mapping the "`"(grave) key to the users actual keyboard layout */
+        if ((keymap[scancode] & SDLK_SCANCODE_MASK) || scancode == SDL_SCANCODE_DELETE /*|| scancode == SDL_SCANCODE_GRAVE*/) {
             continue;
         }
 
         vk = MapVirtualKey(i, MAPVK_VSC_TO_VK);
-        if (vk) {
-            int ch = (MapVirtualKey(vk, MAPVK_VK_TO_CHAR) & 0x7FFF);
+        if (!vk) {
+            continue;
+        }
+
+        /* Always map VK_A..VK_Z to SDLK_a..SDLK_z codes.
+         * This was behavior with MapVirtualKey(MAPVK_VK_TO_CHAR). */
+        //if (vk >= 'A' && vk <= 'Z') {
+        //    keymap[scancode] = SDLK_a + (vk - 'A');
+        //} else {
+        {
+            BYTE keyboardState[256] = { 0 };
+            WCHAR buffer[16] = { 0 };
+            Uint32 *ch = 0;
+            int result = ToUnicode(vk, i, keyboardState, buffer, 16, 0);
+            buffer[SDL_abs(result) + 1] = 0;
+
+            /* Convert UTF-16 to UTF-32/UCS-4 code points */
+            ch = (Uint32 *)SDL_iconv_string("UCS-4-INTERNAL", "UTF-16LE", (const char *)buffer, (SDL_abs(result) + 1) * sizeof(WCHAR));
```

Comment:
The -INTERNAL variants are deprecated, this should probably just use "UCS-4" or "UTF-32LE"

Reply:
fixed